pkg/component/controller/apiendpointreconciler_test.go

```diff
@@ -213,3 +180,26 @@ func verifyEndpointAddresses(t *testing.T, expectedAddresses []string, fakeFacto
 
 	return ep
 }
+
+type fakeResolver struct{}
+
+func (fr fakeResolver) LookupIPAddr(ctx context.Context, host string) ([]net.IPAddr, error) {
+	return []net.IPAddr{
```

Comment:
A bit late to the party, but I'd probably have made sure the correct host was passed by the caller.